pipelines/validate_vset.py

```
@@ -0,0 +1,894 @@
+    if kind == "review_remediation_v1":
+        patch = payload.get("revised_patch")
+    elif kind == "failure_recovery_v1":
+        action = payload.get("recovery_action")
+        patch = action.get("patch") if isinstance(action, Mapping) else None
+    else:
+        patch = payload.get("patch")
+    return patch if isinstance(patch, Mapping) else None
+
+
+def validate_record_with_oracle(
+    record: Any,
```

Ruff (0.16.2)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `record`

(ANN401)